activesupport/lib/active_support/core_ext/hash/except.rb

```diff
@@ -11,7 +11,7 @@ class Hash
   #   @person.update(params[:person].except(:admin))
   def except(*keys)
     slice(*self.keys - keys)
-  end
+  end unless {}.respond_to?(:except)
```

Comment:
It is better to check ruby version here so that this code will be easy to cleanup when the support for older ruby versions will be dropped.

Reply:
We usually check if a feature exists.

https://github.com/rails/rails/blob/d6a90028244f746ac30de7255ba996a6f37f5757/activesupport/lib/active_support/core_ext/symbol/starts_ends_with.rb#L6
https://github.com/rails/rails/blob/92fd1c202bc6cdb945c110b23f9879de0fbd9b0e/activesupport/lib/active_support/core_ext/hash/slice.rb#L25
https://github.com/rails/rails/blob/92fd1c202bc6cdb945c110b23f9879de0fbd9b0e/activesupport/lib/active_support/core_ext/hash/transform_values.rb#L21
https://github.com/rails/rails/blob/92fd1c202bc6cdb945c110b23f9879de0fbd9b0e/activesupport/lib/active_support/core_ext/regexp.rb#L10
https://github.com/rails/rails/blob/e4b6c719cd1df3dd2bf16645dbe7780dd541ec78/activejob/lib/active_job/arguments.rb#L72

Reply:
Well, for exmaple, ruby support `Regex#match?` since 2.4: `https://ruby-doc.org/core-2.4.0/Regexp.html#method-i-match-3F`. It means that https://github.com/rails/rails/blob/92fd1c202bc6cdb945c110b23f9879de0fbd9b0e/activesupport/lib/active_support/core_ext/regexp.rb#L10 would need to be cleaned up already as ruby requirements is `>= 2.5` now, but it will not because there is no reference. 

I saw similar problems 2 years ago where a really crazy hack in routing helpers existed because of problem with ruby 1.9, but no one cleaned it up because it checked if object respond to method instead of ruby version check.

So who knows how many problems like that are in the code.